agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/processors/ExampleExporterTest.java

```diff
@@ -0,0 +1,804 @@
+    }
+
+    @Test
+    public void simpleExcludeRegexTest() {
+        MockExporter mockExporter = new MockExporter();
+        SpanProcessorConfig config = new SpanProcessorConfig();
+        config.exclude = new SpanProcessorIncludeExclude();
+        config.exclude.match_type = "regexp";
+        config.exclude.span_names = Arrays.asList("svc.*");
+        SpanProcessorAction action=new SpanProcessorAction();
+        action.key="testKey";
+        action.action="update";
+        action.value="redacted";
+        List<SpanProcessorAction> actions=new ArrayList<>();
+        actions.add(action);
+        config.actions=actions;
+        SpanExporter exampleExporter = ExampleExporter.create(config, mockExporter);
+
+        Span spanA = OpenTelemetry.getTracer("test").spanBuilder("svcA")
+                .setAttribute("one", "1")
+                .setAttribute("two", 2L)
+                .setAttribute("testKey","testValue")
+                .setAttribute("testKey2","testValue2")
+                .startSpan();
+        Span spanB = OpenTelemetry.getTracer("test").spanBuilder("svcB")
+                .setAttribute("one", "1")
+                .setAttribute("testKey","testValue")
+                .setAttribute("testKey2","testValue2")
+                .startSpan();
+        Span spanC = OpenTelemetry.getTracer("test").spanBuilder("serviceC")
+                .setAttribute("two", 2L)
+                .setAttribute("testKey","testValue")
+                .setAttribute("testKey2","testValue2")
+                .startSpan();
+        Span spanD = OpenTelemetry.getTracer("test").spanBuilder("serviceD")
+                .setAttribute("one", "1")
+                .setAttribute("two", 2L)
+                .setAttribute("testKey","testValue")
+                .setAttribute("testKey2","testValue2")
+                .startSpan();
+
+        List<SpanData> spans = new ArrayList<>();
+        spans.add(((ReadableSpan) spanA).toSpanData());
+        spans.add(((ReadableSpan) spanB).toSpanData());
+        spans.add(((ReadableSpan) spanC).toSpanData());
+        spans.add(((ReadableSpan) spanD).toSpanData());
+
+        exampleExporter.export(spans);
+
+        // verify that resulting spans are filtered in the way we want
+        List<SpanData> result = mockExporter.getSpans();
+        SpanData resultSpanA = result.get(0);
+        SpanData resultSpanB = result.get(1);
+        SpanData resultSpanC = result.get(2);
+        SpanData resultSpanD = result.get(3);
+        assertEquals("testValue",resultSpanA.getAttributes().get("testKey").getStringValue());
+        assertEquals("testValue",resultSpanB.getAttributes().get("testKey").getStringValue());
+        assertEquals("redacted",resultSpanC.getAttributes().get("testKey").getStringValue());
+        assertEquals("redacted",resultSpanD.getAttributes().get("testKey").getStringValue());
+    }
+
+    @Test
+    public void multiIncludeTest() {
```

Comment:
verify that multi attributes are or-ed not and-ed

Reply:
Looks like multi attributes are and-ed.
https://github.com/open-telemetry/opentelemetry-collector/blob/master/processor/README.md#includeexclude-spans